packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -549,7 +550,7 @@ export class Pipeline extends PipelineBase {
       disableInboundStageTransitions: Lazy.any({ produce: () => this.renderDisabledTransitions() }, { omitEmptyArray: true }),
       roleArn: this.role.roleArn,
       restartExecutionOnUpdate: props && props.restartExecutionOnUpdate,
-      pipelineType: props.pipelineType,
+      pipelineType: props.pipelineType ?? (isDefaultV2 ? PipelineType.V2 : undefined),
```

Comment:
nit: seems redundant check since on line 536, you've already used null coalescing operator `??`.